analytical/templatetags/matomo.py

```diff
@@ -0,0 +1,118 @@
+"""
+Matomo template tags and filters.
+"""
+
+from __future__ import absolute_import
+
+from collections import namedtuple
+from itertools import chain
+import re
+
+from django.conf import settings
+from django.template import Library, Node, TemplateSyntaxError
+
+from analytical.utils import (is_internal_ip, disable_html,
+                              get_required_setting, get_identity)
+
+
+# domain name (characters separated by a dot), optional port, optional URI path, no slash
+DOMAINPATH_RE = re.compile(r'^(([^./?#@:]+\.)*[^./?#@:]+)+(:[0-9]+)?(/[^/?#@:]+)*$')
+
+# numeric ID
+SITEID_RE = re.compile(r'^\d+$')
+
+TRACKING_CODE = """
+<script type="text/javascript">
+  var _paq = window._paq || [];
+  %(variables)s
+  %(commands)s
+  _paq.push(['trackPageView']);
+  _paq.push(['enableLinkTracking']);
+  (function() {
+    var u="//%(url)s/";
+    _paq.push(['setTrackerUrl', u+'matomo.php']);
+    _paq.push(['setSiteId', %(siteid)s]);
+    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
+    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
+  })();
+</script>
+<noscript><p><img src="//%(url)s/piwik.php?idsite=%(siteid)s" style="border:0;" alt="" /></p></noscript>
+"""  # noqa
+
+VARIABLE_CODE = '_paq.push(["setCustomVariable", %(index)s, "%(name)s", "%(value)s", "%(scope)s"]);'  # noqa
+IDENTITY_CODE = '_paq.push(["setUserId", "%(userid)s"]);'
+DISABLE_COOKIES_CODE = '_paq.push([\'disableCookies\']);'
+
+DEFAULT_SCOPE = 'page'
+
+MatomoVar = namedtuple('MatomoVar', ('index', 'name', 'value', 'scope'))
+
+
+register = Library()
+
+
+@register.tag
+def matomo(parser, token):
+    """
+    Matomo tracking template tag.
+
+    Renders Javascript code to track page visits.  You must supply
+    your Matomo domain (plus optional URI path), and tracked site ID
+    in the ``MATOMO_DOMAIN_PATH`` and the ``MATOMO_SITE_ID`` setting.
+
+    Custom variables can be passed in the ``matomo_vars`` context
+    variable.  It is an iterable of custom variables as tuples like:
+    ``(index, name, value[, scope])`` where scope may be ``'page'``
+    (default) or ``'visit'``.  Index should be an integer and the
+    other parameters should be strings.
+    """
+    bits = token.split_contents()
+    if len(bits) > 1:
+        raise TemplateSyntaxError("'%s' takes no arguments" % bits[0])
+    return MatomoNode()
+
+
+class MatomoNode(Node):
```

Comment:
1. Make `PiwikNode` empty, make it inherit from this implementation (i.e. `PiwikNode(MatomoNode)`), and have it emit a [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning).